src/qtiCommonRenderer/renderers/interactions/TextEntryInteraction.js

```diff
@@ -72,6 +72,29 @@ function showTooltip($input, theme, message) {
     $input.data('$tooltip').show();
 }
 
+/**
+ * Validate the input for decimal values.
+ *
+ * This function ensures that the input value is either empty or follows
+ * the rules for decimal numbers. It allows numbers with optional
+ * thousands separators (commas) and a mandatory decimal point (dot).
+ *
+ * @param {jQuery} $input
+ */
+function validateDecimalInput ($input) {
+    const value = $input.val();
+    const regex = /^$|^-?\d{1,3}(,\d{3})*(\.\d+)?$|^-?\d+(\.\d+)?$|^-?\d*\.$|^-?\.\d+$/;
+    if (!regex.test(value)) {
+        $input.addClass('invalid');
+        $input.addClass('error');
+        showTooltip($input, 'error', __('Invalid value, use . (dot) for decimal point and , (comma) for thousands separator.'));
```

Comment:
Just a note: the ticket says: `The value is still truncated if submitted as an invalid value.`3,14` is submitted as `3`. But I think we should submit `null`, as we do in Solar? Anyway, this is out of scope in this ticket.

Reply:
In task they said: 'The value is still truncated if submitted as an invalid value.', thats why I didn't change this logic.